src/Document.js

```diff
@@ -167,68 +167,74 @@ function initDocument (document, window) {
       } catch(err) {
         console.warn(err);
       }
+    } else {
+      try {
+        await GlobalContext._runHtml(document, window);
+      } catch(err) {
+        console.warn(err);
+      }
 
-      document.readyState = 'interactive';
-      document.dispatchEvent(new Event('readystatechange', {target: document}));
+      document.dispatchEvent(new Event('DOMContentLoaded', {target: document}));
```

Comment:
This looks like a double `DOMContentLoaded` emit. See https://github.com/webmixedreality/exokit/pull/348/files#diff-614df7e2218ad0ff0e6a77be196d9f7cR163.

This would most likely would corrupt a site listening for this event, wouldn't it?

Reply:
![image](https://user-images.githubusercontent.com/59632/44007996-ad548f14-9e63-11e8-859b-ba9ed6851d38.png)

This diff is hard to read on github, but looking at the screenshot makes it clearer.

The event is emitted only once, either in the `if` or the `else` branch. Probably worth breaking up the function into smaller pieces.